massa-versioning/src/versioning_ser_der.rs

```diff
@@ -905,40 +895,120 @@ mod test {
 
         let mut buf = Vec::new();
         let mip_info_ser = MipInfoSerializer::new();
-        mip_info_ser.serialize(&vi_1, &mut buf).unwrap();
+        mip_info_ser.serialize(&mi_1, &mut buf).unwrap();
 
         let mip_info_der = MipInfoDeserializer::new();
 
-        let (rem, vi_1_der) = mip_info_der.deserialize::<DeserializeError>(&buf).unwrap();
+        let (rem, mi_1_der) = mip_info_der.deserialize::<DeserializeError>(&buf).unwrap();
 
         assert!(rem.is_empty());
-        assert_eq!(vi_1, vi_1_der);
+        assert_eq!(mi_1, mi_1_der);
     }
 
     #[test]
-    fn test_component_state_ser_der() {
-        let st_1 = ComponentState::failed();
-        let st_2 = ComponentState::Started(Started {
-            vote_ratio: Ratio::from_f32(0.9842).unwrap(),
-        });
+    fn test_mip_info_ser_der_err() {
+        // A MIP info with too many MIP Component
+        let mi_1 = MipInfo {
+            name: "MIP-0002".to_string(),
+            version: 2,
+            components: BTreeMap::from([
+                (MipComponent::Address, 1),
+                (MipComponent::KeyPair, 2),
+                (MipComponent::Block, 3),
+                (MipComponent::VM, 4),
+                (MipComponent::FinalStateHashKind, 5),
+                (MipComponent::__Nonexhaustive, 6),
+            ]),
+            start: MassaTime::from_millis(2),
+            timeout: MassaTime::from_millis(5),
+            activation_delay: MassaTime::from_millis(2),
+        };
 
-        let mut buf = Vec::new();
-        let state_ser = ComponentStateSerializer::new();
-        state_ser.serialize(&st_1, &mut buf).unwrap();
+        {
+            let mut buf = Vec::new();
+            let mip_info_ser = MipInfoSerializer::new();
+            mip_info_ser.serialize(&mi_1, &mut buf).unwrap();
+
+            let mut mip_info_der = MipInfoDeserializer::new();
+            // Allow only a max of 2 components per MIP info
```

Comment:
Here 2 is excluded? We should change the comment

Reply:
Changed